x-pack/plugins/transform/public/app/common/request.ts

```diff
@@ -188,10 +194,16 @@ export const getCreateTransformSettingsRequestBody = (
   transformDetailsState: Partial<StepDetailsExposedState>
 ): { settings?: PutTransformsRequestSchema['settings'] } => {
   const settings: PutTransformsRequestSchema['settings'] = {
-    ...(transformDetailsState.transformSettingsMaxPageSearchSize
+    // conditionally add optional max_page_search_size, skip if default value
+    ...(transformDetailsState.transformSettingsMaxPageSearchSize &&
+    transformDetailsState.transformSettingsMaxPageSearchSize !==
+      DEFAULT_TRANSFORM_SETTINGS_MAX_PAGE_SEARCH_SIZE
       ? { max_page_search_size: transformDetailsState.transformSettingsMaxPageSearchSize }
       : {}),
-    ...(transformDetailsState.transformSettingsDocsPerSecond
+    // conditionally add optional docs_per_second, skip if default value
+    ...(transformDetailsState.transformSettingsDocsPerSecond &&
+    transformDetailsState.transformSettingsDocsPerSecond !==
+      DEFAULT_TRANSFORM_SETTINGS_DOCS_PER_SECOND
```

Comment:
but what if the user sets the default value manually? I mean explicitly

Reply:
Currently the input fields in the creation wizards and the wizard's state get populated with the default values on load, so they are already there explicitly. If a user doesn't change it, it will now be skipped from being populated it in the request. Having it the other way around would require quite some more changes how we handle form state, so I'd prefer to keep it like this for now with this more minimal fix.

Reply:
I added an item to this meta issue for the transform plugin's form states about the issue of populating the whole form with default values: https://github.com/elastic/kibana/issues/77604